src/modules/commander/Commander.cpp

```diff
@@ -3950,19 +3952,26 @@ void Commander::data_link_check(bool &status_changed)
 
 	// AVOIDANCE SYSTEM state check (only if it is enabled)
 	if (_obs_avoid.get()) {
+		static bool print_once = false;
 
 		//if avoidance never started
 		if (_datalink_last_heartbeat_avoidance_system == 0 && hrt_elapsed_time(&_avoidance_system_not_started) > 5_s) {
 			_avoidance_system_not_started = hrt_absolute_time();
-			mavlink_log_critical(&mavlink_log_pub, "AVOIDANCE SYSTEM DID NOT START");
+			mavlink_log_critical(&mavlink_log_pub, "WAITING FOR AVOIDANCE SYSTEM");
 		}
 
 		//if heartbeats stop
 		if ((_datalink_last_heartbeat_avoidance_system > 0)
 		    && (hrt_elapsed_time(&_datalink_last_heartbeat_avoidance_system) > 5_s) &&
 		    (hrt_elapsed_time(&_avoidance_system_lost) > 5_s)) {
 			_avoidance_system_lost = hrt_absolute_time();
```

Comment:
I know that it's not from this PR but a couple of lines above `_avoidance_system_lost`  is a boolean

Reply:
I got tricked in the [header definition](https://github.com/PX4/Firmware/blob/99f8e38de02805723ce725c4299e35f5ee8cc4fa/src/modules/commander/Commander.hpp#L189) where is declared as uint64_t. I am gonna try tomorrow by cherry picking your commits.